site/docs/v1/tech/lambdas/client-credentials-jwt-populate.adoc

```diff
@@ -0,0 +1,87 @@
+description: An overview of the Client Credentials JWT Populate lambda
+---
+
+:sectnumlevels: 0
+
+== Client Credentials JWT Populate lambda
+
+If you would like to augment the claims provided in the JWT before it has been signed you can specify a lambda in the JWT configuration. This lambda will be invoked prior to the token being signed and issued as a result of the client credentials grant, on behalf of an Entity.
+
+When you create a new lambda using the FusionAuth UI we will provide you an empty function for you to implement. If you are using the API to create the lambda you will need to ensure your function has the following signature:
+
+[source,javascript]
+----
+function populate(jwt, recipientEntity, targetEntities, permissions) {
+  // Lambda code goes here
+}
+----
+
+This lambda must contain a function named `populate` that takes four parameters. The parameters that the lambda is passed are:
+
+* `jwt` - the claims object to be signed and return as the JWT payload
+* `recipientEntity` - the Recipient Entity.  Example below.
+* `targetEntities` - the Target Entities. Example below.
+* `permissions` - the permissions assigned to the Entity. Example below.
+
+The `recipientEntity` and `targetEntities` objects are well documented here in the link:/docs/v1/tech/apis/entity-management/entity-types/[Permissions & Entity Types APIs] and link:/docs/v1/tech/apis/entity-management/entities/[Entities API] documentation. The JWT object is a JavaScript object containing the JWT payload. See link:/docs/v1/tech/oauth/tokens/[OpenID Connect & OAuth 2.0 Token].
+
+You may add or modify anything in the `jwt` object. However, you may not modify the header keys or values of the JWT. FusionAuth also protects certain reserved claims. The following claims are considered reserved and modifications or removal will not be reflected in the final JWT payload:
+
+- `exp`
+- `iat`
+- `sub`
+
+== Assigning the lambda
+
+Once a lambda is created, you must assign it. See the OAuth tab in the under tenants.
+
+== Example lambda
+
+There is not a default Client Credentials Populate lambda provided by FusionAuth. The following is an example lambda you can use as a starting point.
+
+[source,javascript]
+----
+// Using the two Entity and permission parameters you can populate a client credentials grant JWT.
+function populate(jwt, recipientEntity, targetEntities, permissions) {
+  //  When writing a lambda we've added a few helpers to make life easier.
+  //  console.info('Hello World');         # This will create an EventLog of type Information
+  //  console.error('Not good.');          # This will create an EventLog of type Error
+  //  console.debug('Step 42 completed.'); # This will create an EventLog of type Debug
+   //
+  //  To dump an entire object to the EventLog you can use JSON.stringify, for example:
+  //  console.info(JSON.stringify(user));
+
+  // Happy coding! Populate your JWT here.
+
+  console.info('Hello World!');
```

Comment:
This is probably ok, but since we don't ship a default, the purpose of an example would be to show them some real code. So we could build a workable example here of doing something with the JWT.